app/controllers/app.py

```diff
@@ -1292,3 +1292,127 @@ def post(self, project_id, app_id):
             return dict(status='fail', message='No values found'), 404
 
         return dict(status='success', data=dict(values=network_data_list)), 200
+
+
+class AppLogsView(Resource):
+    @jwt_required
+    def post(self, project_id, app_id):
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        logs_schema = PodsLogsSchema()
+        logs_data = request.get_json()
+
+        validated_query_data, errors = logs_schema.load(logs_data)
+
+        if errors:
+            return dict(status='fail', message=errors), 400
+
+        project = Project.get_by_id(project_id)
+
+        if not project:
+            return dict(
+                status='fail',
+                message=f'project {project_id} not found'
+            ), 404
+
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
+            return dict(status='fail', message='unauthorised'), 403
+
+        # Check app from db
+        app = App.get_by_id(app_id)
+
+        if not app:
+            return dict(
+                status='fail',
+                message=f'app {app_id} not found'
+            ), 404
+
+        cluster = project.cluster
+        if not cluster:
+            return dict(status='fail', message="Invalid Cluster"), 500
+
+        kube_host = cluster.host
+        kube_token = cluster.token
+        kube_client = create_kube_clients(kube_host, kube_token)
+
+        namespace = project.alias
+        deployment = f'{app.alias}-deployment'
+
+        tail_lines = validated_query_data.get('tail_lines', 100)
+        since_seconds = validated_query_data.get('since_seconds', 86400)
+        timestamps = validated_query_data.get('timestamps', False)
+
+        ''' Get Replicas sets'''
+        replicas = kube_client.appsv1_api.list_namespaced_replica_set(
+            namespace).to_dict()
+        replicasList = []
+        for replica in replicas['items']:
+            name = replica['metadata']['name']
+            if name.startswith(deployment):
+                replicasList.append(name)
+
+        ''' get pods list'''
+        pods = kube_client.kube.list_namespaced_pod(namespace)
+        podsList = []
+        failed_pods = []
+        for item in pods.items:
+            item = kube_client.api_client.sanitize_for_serialization(item)
+            pod_name = item['metadata']['name']
+
+            try:
+                status = item['status']['conditions'][1]['status']
+            except:
+                continue
+
+            for replica in replicasList:
+                if pod_name.startswith(replica):
+                    if status == 'True':
+                        podsList.append(pod_name)
+                    else:
+                        failed_pods.append(pod_name)
+                    continue
```

Comment:
Why do we need this `continue` statement?

Reply:
Because if the pod has been added, I don't  want the same pod to be added again if its deployment starts with the same pod name